### ISSUE TITLE
Implement streams as with node-canvas

### ISSUE THREAD

**Issue by GreenFlag31** [Brooooooklyn/canvas#963]
Hi.

Your solution has been choosed as replacement for pdfjs-dist of node-canvas. However, it does not implement streams (nodejs) as with node canvas : see [here](https://github.com/Automattic/node-canvas?tab=readme-ov-file#canvascreatepngstream).
Implementing streams improves memory consumption and is not slower according to my tests (tested to convert a PDF into an image with/without streams).

Would it be something you could consider?

**Comment by santiq** [comment]
I know its not technically the same, but I'm using this as a quick patch.

```
const { Readable } = require('stream');
const canvasStream = Readable.from(canvas.toBuffer('image/png'));
```

**Comment by GreenFlag31** [comment]
Yep, this is possible, but this is like paying 2 times to cost of rendering. canvas.toBuffer(...) already has a cost. If you wrap it inside a stream, it means paying 2 times to cost of rendering. The goal is to use a stream instead of canvas.toBuffer().
